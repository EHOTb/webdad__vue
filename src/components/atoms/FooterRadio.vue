<template>
  <li
    :class="[$style.radio, { [$style.active]: isActive }]"
    @click="$emit('footerTab')"
  >
    {{ name }}
  </li>
</template>

<script>
export default {
  name: "FooterRadio",

  props: {
    name: String,
    isActive: {
      type: Boolean,
      default: false,
    },
  },
};
</script>

<style lang="scss" module>
@import "../../assets/styles/var.scss";

.radio {
  display: flex;
  margin-right: 1vw;
  align-items: center;
  align-items: flex-end;
  padding-top: 1rem;
}



.active {
  border: 0.15rem solid $main-col-42;
  border-radius: 1rem;
  padding: 0.2rem 0.6rem;
  box-shadow: 0.2rem 0.2rem 0.3rem $main-col;
  transition: 0.4s;
}
</style>
