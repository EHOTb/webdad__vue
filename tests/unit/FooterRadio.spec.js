import { mount } from "@vue/test-utils";
import FooterRadio from "../../src/components/atoms/FooterRadio";

describe("Компонент FooterRadio", () => {
  it("Snapshot FooterRadio", () => {
    const wrapper = mount(FooterRadio);
    expect(wrapper.element).toMatchSnapshot();
  });
  it("emit события", () => {
    const wrapper = mount(FooterRadio);
    wrapper.find("li").trigger("click");
    expect(wrapper.emitted("changeFilter")).toBeTruthy();
  });
  it("props", () => {
    const wrapper = mount(FooterRadio, {
      propsData: {
        name: "test",
      },
    });
    expect(wrapper.text()).toBe("test");
  });
});
