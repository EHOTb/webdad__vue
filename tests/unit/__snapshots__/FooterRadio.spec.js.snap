// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Компонент FooterRadio Snapshot FooterRadio 1`] = `
<li
  class="radio"
>
  
  

</li>
`;
